spec/base_spec.rb

```diff
@@ -42,6 +42,17 @@ class BasicCloud < AssetCloud::Base
     lambda { @fs['./test']   }.should raise_error(AssetCloud::IllegalPath)
   end
 
+  it "should raise error when using with minus relative even after another directory" do
+    lambda { @fs['test/../../test']   }.should raise_error(AssetCloud::IllegalPath)
```

Comment:
Because test/../whatever wouldn't actually be a problem.  That said, it's worth testing since it should still be invalid in my opinion.